### DISCUSSION TITLE
Unable to Calculate KDM0 and Phenoage0 in NHANES 2011-2014

### DISCUSSION THREAD

**Discussion by HengWangJia-LiFan** [dayoonkwon/BioAge#38] [Q&A]
Thank you for your package and guidance, which have been extremely helpful for our research. We are using your package to calculate the biological age for NHANES 2011-2014. Since CRP was not measured during these years, we have excluded CRP and used the remaining 11 variables mentioned in the example. However, when we extracted the data for 2011-2014, neither kdm0 nor phenoage0 was calculated. We are unable to determine the issue. Any guidance you can provide on this matter would be greatly appreciated.

```r

``` hd = hd_nhanes(biomarkers=c("albumin","alp","totchol","lncreat","hba1c","sbp","bun","uap","lymph","mcv","wbc"))
kdm = kdm_nhanes(biomarkers=c("albumin","alp","totchol","lncreat","hba1c","sbp","bun","uap","lymph","mcv","wbc"))
phenoage = phenoage_nhanes(biomarkers=c("albumin_gL","alp","totchol","lncreat_umol","hba1c","sbp","bun","uap","lymph","mcv","wbc"))

**Comment by dayoonkwon** [comment]
`kdm0` and `phenoage0` are pre-computed values from the original published algorithms, not from your custom biomarker set. They require specific biomarkers that aren't available in all NHANES waves:

- `kdm0` requires spirometry (FEV) → only available in 2007-2012
- `phenoage0` requires CRP → only available in 1999-2010 and 2015-2018

Since CRP is missing in 2011-2014, `phenoage0` will be NA for those participants. Your custom measures (`kdm`, `phenoage`, `hd`) using the 11 biomarkers you specified should still compute fine for 2011-2014.
